src/attacks.rs

```diff
@@ -409,8 +407,18 @@ fn append_removal(
 // of the given node. Size of the radius is given radius. It corresponds to the
 // under-specified function "UpdateNodesInRadius" in algo. 6 of
 // https://eprint.iacr.org/2018/944.pdf
-fn update_radius_set(g: &Graph, node: usize, inradius: &mut HashSet<usize>, radius: usize) {
-    let add_direct_nodes = |v: usize, closests: &mut HashSet<usize>, _: &HashSet<usize>| {
+// NOTE: The `radius` shouldn't change across calls for the same `inradius` set,
+// that is, if we already have a node in `inradius` then we won't look for it
+// again because we assume we already found all its closest nodes within a
+// specified `radius` (if the `radius` increased across calls we would be missing
+// nodes that were farther away in comparison to earlier calls).
+fn update_radius_set(g: &Graph, node: usize, inradius: &mut NodeSet, radius: usize) {
+    let mut closests: Vec<Node> = Vec::with_capacity(radius * 10);
+    // FIXME: We should be able to better estimate the size of this scratch
+    //  vector based on the `radius` and the average degree of the nodes.
+    let mut tosearch: Vec<Node> = Vec::with_capacity(closests.capacity());
```

Comment:
We can already provide a lower bound because we know the number of parents: something like `1 + radius^out-degree`. 
Note that this would require getting more information on the graph, and this depends on the DRGAlgo. 
Maybe we can add a function on DRGAlgo that returns this out-degree. 
If we can average out the number of children, then we can update the vector size to take that into account.